src/resolve/resolved-dns-cache.c

```diff
@@ -54,6 +54,7 @@ struct DnsCacheItem {
         LIST_FIELDS(DnsCacheItem, by_key);
 
         bool shared_owner;
+        usec_t ttl_expiry;        /* The key is for storing the time when the TTL set to expire. */
 };
```

Comment:
i'd prefer if we'd store two time-values `until_valid`, `until_stale` or so, where the first one is the time the entry remains fully valid by the indicated TTL and the second one indicates the time it remains "vaguely valid" as per the stale rules.

I think that would simplify things a lot, as the two timeouts are then similarly named, and behave similarly, we just use either one or the other timeout depending on context.

Reply:
Renamed `ttl_expiry `to `until_valid`.